src/fastcs/backends/epics/ioc.py

```diff
@@ -112,13 +114,34 @@ def _create_and_link_attribute_pvs(pv_prefix: str, mapping: Mapping) -> None:
         for attr_name, attribute in single_mapping.attributes.items():
             pv_name = attr_name.title().replace("_", "")
             _pv_prefix = ":".join([pv_prefix] + path)
+            full_pv_name_length = len(f"{_pv_prefix}:{pv_name}")
+
+            if full_pv_name_length > EPICS_MAX_NAME_LENGTH:
+                attribute.enabled = False
+                print(
+                    f"Not creating PV for {attr_name} for controller"
+                    f" {single_mapping.controller.path} as full name would exceed"
+                    f" {EPICS_MAX_NAME_LENGTH} characters"
+                )
+                continue
 
             match attribute:
                 case AttrRW():
-                    _create_and_link_read_pv(
-                        _pv_prefix, f"{pv_name}_RBV", attr_name, attribute
-                    )
-                    _create_and_link_write_pv(_pv_prefix, pv_name, attr_name, attribute)
+                    if full_pv_name_length > (EPICS_MAX_NAME_LENGTH - 4):
+                        if full_pv_name_length <= EPICS_MAX_NAME_LENGTH:
```

Comment:
I don't think the else of this case is reachable, because of the check [here](https://github.com/DiamondLightSource/FastCS/pull/56/files#diff-5cf291216194a03497811c649effa3cf60205d4587ec541263ec5a1f4d3a5fbeR119), so it could be taken out.

Reply:
I think the logic is correct, though maybe a little ugly. It should only print the warning if we have a PV of length 56 < x <= 60, which would be sufficiently short if we didn't append "_RBV". "continue" gets called above for PVs of length x > 60
The else case should be reached when we have a length x <= 56.

The new test in test_ioc.py tests both these cases I believe.

Reply:
Sorry, I mean the else of `if full_pv_name_length <= EPICS_MAX_NAME_LENGTH` not `if full_pv_name_length > (EPICS_MAX_NAME_LENGTH - 4)`.

Above you are checking `full_pv_name_length > EPICS_MAX_NAME_LENGTH` and continuing if true, so  if it gets past that `full_pv_name_length <= EPICS_MAX_NAME_LENGTH` will always be true here.

Reply:
Ahh you're right, not sure how I missed that.

Reply:
Cool. Could you update the tests to cover those two missing lines in src/fastcs/attributes.py and then I think we're good to merge.

Reply:
Assuming you meant to cover the case where Commands with long names get disabled, have pushed that